Feat/backend

Lem0nTree opened the pull request:
- Added backend Routes
- Added database schema
- Added openAI api to trained model
- Added NEAR conversion to USD , cached 12 hours

urataps commented on app/.claudeignore:
Bro why do we need .claudeignore files? Is this built-in feature of cursor?

Lem0nTree replied:
Kind of , is an extension i use in cursor to sync claude projects

urataps commented on app/.claudesync/config.local.json:
What this file does sir?

Lem0nTree replied:
can add it to ignore file

urataps commented on app/src/app/layout.tsx:
```diff
@@ -19,15 +20,17 @@ export default function RootLayout({
+              {children}
+            </TRPCReactProvider>
+          </main>
+        </ToastProvider>
```
ToastProvider does not have to be a wrapper can be a simple <ToastProvider/> at the end

Lem0nTree replied:
I tried without wrapping but is giving me error

urataps commented on app/src/server/api/routers/strategy.ts:
```diff
@@ -0,0 +1,229 @@
+        });
+
+        // Convert NEAR amount to number
+        const nearAmount = parseFloat(input.neat);
```
input.neat?

Lem0nTree replied:
fixed typo

urataps commented on app/src/server/api/routers/strategy.ts:
I'm proud of you sir, this is top notch big brain backend (no joke)

Lem0nTree commented:
- Added skeleton while loading
- Added dynamic icons for chains and protocol
- Added dynamic data in the investment card
- Added tooltip on hover
- Added custom chain colors
- Improved protocols icons
